### PR TITLE
Introduce TCPSettings in Hostrule for ParentVS configuration

### PR DESCRIPTION
**Pull request by chauhanshubham**
This commit introduces tcpSettings section that allows for
parent VS tcp settings like custom ports to be opened as part
of the virtual service, or a static load balancer IP to be
assigned to the VS. Includes a few validaitons around the
new fields too.

### CODE REVIEW

**Comment by chauhanshubham** [comment]
This PR does not cover AKO behaviour when a loadBalancerIP which leads to a VSVIP Post/Put error/failure is provided.

**Comment by chauhanshubham** [comment]
build ako

**Comment by chauhanshubham** [comment]
build ako

File: tests/integrationtest/l4_service_test.go [hunk 1]
```diff
@@ -962,46 +961,3 @@ func TestInfraSettingChangeMapping(t *testing.T) {
 	TeardownAviInfraSetting(t, settingName2)
 	TearDownTestForSvcLB(t, g)
 }
-
-func TestBootupServiceLBStatusPersistence(t *testing.T) {
```
Comment (akshayhavile) [inline tests/integrationtest/l4_service_test.go:966]: Are we covering this use case somewhere else?

File: ako-operator/helm/ako-operator/crds/ako.vmware.com_hostrules.yaml [hunk 2]
```diff
@@ -77,6 +77,20 @@ spec:
...
+                      listeners:
+                        items:
+                          properties:
+                            port:
+                              type: integer
+                            enableSSL:
+                              type: boolean
+                          type: object
+                        type: array
+                      loadBalancerIP:
```
Comment (akshayhavile) [inline ako-operator/helm/ako-operator/crds/ako.vmware.com_hostrules.yaml:91]: Should we include IP type also v4 /V6?

Comment (akshayhavile) [reply ako-operator/helm/ako-operator/crds/ako.vmware.com_hostrules.yaml:91]: Can user specify public ip over here for cloud deployments?

Comment (akshayhavile) [reply ako-operator/helm/ako-operator/crds/ako.vmware.com_hostrules.yaml:91]: How will it behave if EnablepublicIP is true and we have specified IP over here?

Comment (chauhanshubham) [reply ako-operator/helm/ako-operator/crds/ako.vmware.com_hostrules.yaml:91]: We don't have ipv6 support yet, later we can possibly auto detect it from the field.
Not sure about the behaviour but programmatically, handling enablePublicIP and static IP is mutually exclusive.

File: tests/integrationtest/l4_service_test.go (hunk 1 above)
Comment (chauhanshubham) [reply tests/integrationtest/l4_service_test.go:966]: yes bootup scenarios are covered in the functional tests.

**Review by akshayhavile — APPROVED** [review]
LGTM.

**Review by monotosh-avi — APPROVED** [review]

**Pull request merged**